dev/integration_tests/ios_platform_view_tests/ios/PlatformViewUITests/PlatformViewUITests.m

```diff
@@ -56,4 +68,31 @@ - (void)testPlatformViewFocus {
   XCTAssertFalse(flutterTextField.flt_hasKeyboardFocus);
 }
 
+- (void)testPlatformViewZOrder {
+  XCUIElement *entranceButton = self.app.buttons[@"platform view z order test"];
+  XCTAssertTrue([entranceButton waitForExistenceWithTimeout:kStandardTimeOut], @"The element tree is %@", self.app.debugDescription);
+  [entranceButton tap];
+
+  XCUIElement *showAlertButton = self.app.buttons[@"Show Alert"];
+  XCTAssertTrue([showAlertButton waitForExistenceWithTimeout:kStandardTimeOut]);
+
+  [showAlertButton tap];
+
+  XCUIElement *platformButton = self.app.buttons[@"platform_view[0]"];
+  XCTAssertTrue([platformButton waitForExistenceWithTimeout:kStandardTimeOut]);
+  XCTAssertTrue([platformButton.label isEqualToString:@"Initial Button Title"]);
+
+  // The `app.otherElements` query fails to query `platform_view[1]`,
+  // which is likely a bug in XCUITest since it is present in the UI debugger.
+  // So set a manual delay here (must be longer than the corresponding
+  // delay to show background view on the dart side).
```

Comment:
Corresponding to which delay?

Reply:
let me update to be more clear.

Reply:
So we discovered that it is not a bug rather a correct behavior of the framework. So we should probably explain that behavior here.